src/main/scala/beam/utils/protocolvis/VisualizingApp.scala

```diff
@@ -0,0 +1,124 @@
+package beam.utils.protocolvis
+
+import beam.utils.protocolvis.Extractors.{AllMessages, ByPerson, ExtractorType}
+import com.typesafe.scalalogging.StrictLogging
+
+import java.io.File
+import java.nio.file.{Files, Path, Paths}
+import java.time.LocalDateTime
+import scala.annotation.tailrec
+import scala.util.{Failure, Success, Try}
+
+/* Converts beam actor messages to a sequence diagram. This messages are written when the beam config file contains
+akka.actor.debug.receive=true
+In this case beam produces files like 0.actor_messages_0.csv.gz in each iteration folder.
+
+Order of the messages is not always right because each actor has its own message queue.
+
+  Parameters:
+--input output/sf-light/run_name__2021-03-29_19-04-50_vnh/ITERS/it.0
+--output docs/uml/choose_mode.puml
+--force
+--person-id 010900-2012001379980-0-560057
+
+ input directory where to read message files,
+ output the output file,
+ force allows to overwrite the output without prompt
+ person-id the person id which the message sequence should be generated for
+ */
+object VisualizingApp extends StrictLogging {
+
+  def main(args: Array[String]): Unit = {
+    parseArgs(args) match {
+      case Some(cliOptions) =>
+        val confirm = confirmOverwrite(cliOptions.output, cliOptions.forceOverwriting)
+        val extractorType = if (cliOptions.personId.isEmpty) AllMessages else ByPerson(cliOptions.personId)
+        if (confirm) doJob(cliOptions.input, cliOptions.output, extractorType) else println("Exiting...")
+      case None => System.exit(1)
+    }
+  }
+
+  private def doJob(inputFile: Path, output: Path, extractorType: ExtractorType): Unit = {
+    import java.time.temporal.ChronoUnit.SECONDS
+    val startTime = LocalDateTime.now()
+    logger.info(s"Start reading from $inputFile")
+    val (csvStream, closable) = MessageReader.readData(inputFile)
+    val extractor = Extractors.messageExtractor(extractorType)
```

Comment:
This all should be in a try{...}catch{...}finally{closable.close()} and would likely read easier and reduce the code.